cpp/CMakeLists.txt

```diff
@@ -92,7 +92,7 @@ include_directories(src)
 include_directories(${CMAKE_BINARY_DIR}/src)  # for format.pb.{h/cc}
 
 function(add_lance_test test_name)
-    if (NOT CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
Change CMakeFile to allow python side to build `ReleaseWithDebugInfo` mode , easier to see call stack.